recipes/b2/portable/conanfile.py

```diff
@@ -53,49 +57,115 @@ def validate(self):
             raise ConanInvalidConfiguration(
                 "Option toolset 'cxx' and 'cross-cxx' requires 'use_cxx_env=True'")
 
+    def layout(self):
+        conan.tools.layout.basic_layout(self, src_folder="root")
```

Comment:
an aside....

Since there is no enforcement so this is totally fine personally, I know its not how we usually import things

But if anyone has objects I'd be very happy to accept a linter PR that enforces our syntax :)

Reply:
I remember there was some issue/limitation about conan imports, like not all of the below was working the same:
```
from conan.tools.layout import basic_layout
basic_layout(self)
...
from conan.tools import layout
layout.basic_layout(self)
...
from conan import tools
tools.layout.basic_layout(self)
...
import conan
conan.tools.layout.basic_layout(self)
```

from python's point of view, all of the above should have the same effect. as long as it works and passes CI, either of them should be allowed.

Reply:
The second lever is kot recommended, those are considered private. I can't find the docs this evening.

It's always valid python to drill down, don't import  `tools` the rest are fair game if they are public

Reply:
this one `conan.tools.layout.basic_layout(self)` is most safe, I would say, as other modules may also expose `basic_layout`, or we can easily encounter naming collisions with other simple names (like `copy`, `get`, etc.)